### DISCUSSION TITLE
[@types/node] NodeJS.ProcessEnv cannot be used in place of Record<string, string | undefined>

### DISCUSSION THREAD

**Discussion by adrian-gierakowski** [DefinitelyTyped/DefinitelyTyped#62806] [Issues with a @types package]
Given

`@types/node@npm:16.11.68`

and: 

```sh
> tsc --version
Version 4.7.4
> node --version
v16.15.0
> yarn info |
```

I'm encountering the following problem:

```ts
type MyPartialEnv = {
  MY_ENV_VAR?: string
}

const getMyEnvVar = (env: MyPartialEnv): string | undefined => env.MY_ENV_VAR

// This is ok, but not ideal
getMyEnvVar(process.env as MyPartialEnv)

// Error, see below
getMyEnvVar(process.env)
```

here's the error I get:

```
11:13  error   Argument of type 'ProcessEnv' is not assignable to parameter of type 'PartialEnv'. ​typescript:2345
                    Property 'MY_ENV_VAR' is missing in type 'ProcessEnv' but required in type 'PartialEnv'.
```

If I remove the following line from ProcessEnv declaration, the error disappears:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e70b22789742c3bbb5d91f61be2dc7707ee1ec5b/types/node/v16/ts4.8/process.d.ts#L115

The reason I want to use `MyPartialEnv` instead of `NodeJS.ProcessEnv` as the type of `getMyEnvVar`'s argument is that with the former my IDE will tell me which env vars are expected, which is useful when manually constructing the argument in tests.

[a comment by an automated account is not shown]

**Comment by yoursunny** [comment]
You can use [@strattadb/environment](https://www.npmjs.com/package/@strattadb/environment) library to validate environment variables and return them as a properly typed object.

**Comment by adrian-gierakowski** [comment]
Thanks for the tip @yoursunny!

However runtime parsing\validation of env vars is orthogonal the the issue I have reported.

The problem is that given current typings one cannot simply use process.env (without type assertion) in places where Record<string, string> could be, which is quite annoying given that is a correct type for process.env. 

I’m not quite sure why TZ should be singled out as the only env var listed explicitly, so I’d like to propose that it is removed. Or maybe this is an issue which should be raised with typescript developers as I don’t see why listing adding an explicit key should break things this way.

**Comment by peterblazejewicz** [comment]
so this is supposed pattern to use (via augmentation):
```ts
declare namespace NodeJS {
    interface ProcessEnv {
        MY_ENV_VAR?: string;
    }
}
// later
const getMyEnvVar = (env: NodeJS.ProcessEnv) => env.MY_ENV_VAR;
getMyEnvVar(process.env);
```
